projects/storefrontlib/cms-components/cart/validation/cart-validation-warnings-state.service.ts

```diff
@@ -1,23 +1,25 @@
-import { Injectable } from '@angular/core';
-import { ReplaySubject, Subject } from 'rxjs';
+import { Injectable, OnDestroy } from '@angular/core';
+import { ReplaySubject, Subject, Subscription } from 'rxjs';
 import { take, takeUntil, tap, withLatestFrom } from 'rxjs/operators';
 import { RoutingService, CartModification } from '@spartacus/core';
 
 @Injectable({
   providedIn: 'root',
 })
-export class CartValidationWarningsStateService {
+export class CartValidationWarningsStateService implements OnDestroy {
   constructor(protected routingService: RoutingService) {
     this.routingService
       .getRouterState()
       .pipe(take(1))
       .subscribe(
         (routerState) => (this.navigationIdCount = routerState.navigationId)
       );
+    this.subscription.add(this.checkForValidationResultClear$.subscribe());
   }
   NAVIGATION_SKIPS = 2;
   navigationIdCount = 0;
 
+  subscription = new Subscription();
```

Comment:
how about setting it as private or at least protected? I think mostly we use private for it